packages/components/organisms/f-form/src/components/Form.vue

```diff
@@ -25,46 +47,71 @@
 
 
 <script>
+import { VueGlobalisationMixin } from '@justeat/f-globalisation';
 import FButton from '@justeat/f-button';
 import FormField from '@justeat/f-form-field';
-import { globalisationServices } from '@justeat/f-services';
+import ErrorMessage from '@justeat/f-error-message';
+import FormValidationMixin from '../mixin/formValidation.mixin';
 import tenantConfigs from '../tenants';
 import { DEFAULT_BUTTON_TEXT, FORM_EVENTS, PROP_VALIDATION_MESSAGES } from '../constants';
+import { FormFieldClass, FormFieldErrorClass } from '../services/formField';
 
 export default {
     components: {
         FButton,
+        ErrorMessage,
         FormField
     },
 
+    mixins: [
+        FormValidationMixin,
+        VueGlobalisationMixin
+    ],
+
     props: {
         formData: {
             type: Object,
             required: true
         },
 
+        locale: {
+            type: String,
+            default: 'en-GB'
+        },
+
         isFormSubmitting: {
             type: Boolean,
             default: false
         }
     },
 
     data () {
-        const locale = globalisationServices.getLocale(tenantConfigs, this.locale, this.$i18n);
-        const localeConfig = tenantConfigs[locale];
-
         return {
-            copy: { ...localeConfig }
+            tenantConfigs
         };
     },
 
     computed: {
         formFields () {
             const formFields = {};
 
-            this.formData.formFields.forEach(field => {
-                formFields[field.name] = field.value;
-            });
+            if (this.formData.formFields?.length) {
+                this.formData.formFields.forEach(field => {
+                    formFields[field.name] = field.value;
+                });
+            }
+
+            return formFields;
+        },
+
+        formFieldData () {
+            const formFields = [];
+
+            if (this.formData.formFields?.length) {
+                this.formData.formFields.forEach(field => {
+                    formFields.push(this.createField(field));
+                });
+            }
 
             return formFields;
```

Comment:
Nice little one-liner I think you should be able to use:
```suggestion
            return this.formData.formFields.map(field => this.createField(field));
```

Reply:
😮 Thank you!